packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -105,6 +118,7 @@ class Dismissible extends StatefulWidget {
     this.background,
     this.secondaryBackground,
     this.confirmDismiss,
+    this.shouldTriggerDismiss,
     this.onResize,
     this.onUpdate,
```

Comment:
Can onUpdate not be used for this? It is called when the dismissible is dragged. I am reading through this widget (having not worked on it a lot before) and noticing we have a lot of different callbacks. If we can repurpose onUpdate I think it could be clearer.

Reply:
What if we added a method for folks to programmatically trigger the dismiss instead? Then here in onUpdate folks could trigger it if they wanted. It would be a smaller amount of public API, and maybe better handle possible conflicts with other parts of the API that already exist.

Reply:
`onUpdate` is currently called using a listener on the AnimationController, so it is called multiple times when dragging the widget. It is not part of any logic right now, it's only reacting to animation changes.
`shouldTriggerDismiss` is only called once per dismiss attempt, when the drag gesture ends (where the flinging detection logic is also implemented)

While it seems like a good idea to add a return value to `onUpdate`, I don't think we will be able to tell precisely if the drag has ended when calling the `onUpdate` callback. Also, `onUpdate` is usually called at least one more time after `shouldTriggerDismiss` is called, when the animation resets or goes out of screen (dismiss).

With that in mind, I would also keep `shouldTriggerDismiss`. I don't think we can make it work with a single callback